File: D7SMS-Ruby/test/controllers/test_api_controller.rb
# d7_sms
#
# https://d7networks.com ).

require_relative 'controller_test_base'

class APIControllerTests < ControllerTestBase
  # Called only once for the class before any test has executed
  def self.startup
    self.controller = @@api_client.client
  end

  # Check Balance
  def test_balance()

    # Perform the API call through the SDK function
    assert_raise('APIException'){ 
      self.class.controller.get_balance()
    }

    # Test response code
    assert_equal(@response_catcher.response.status_code, 500)
  end

  # Send SMS  to a recipient using D7 SMS Gateway
  def test_send_sms()
    # Parameters for the API call
    options = {}
    options['body'] = SendSMSRequest.from_hash(JSON.parse('{"to":971562316353,"from":"SignSMS","content":"Send single SMS Testing"}'))
    options['content_type'] = 'application/json'
    options['accept'] = 'application/json'

    # Perform the API call through the SDK function
    assert_raise('APIException'){ 
      self.class.controller.create_send_sms(options)
    }

    # Test response code
    assert_equal(@response_catcher.response.status_code, 500)
  end

  # Send SMS  to multiple recipients using D7 SMS Gateway
  def test_bulk_sms()
    # Parameters for the API call
    body = BulkSMSRequest.from_hash(JSON.parse(
      '{"messages":[{"to":["971562316353","971562316354","971562316355"],"content'\
      '":"Same content goes to three numbers","from":"SignSMS"}]}'
      ))
    content_type = 'application/json'
    accept = 'application/json'

    # Perform the API call through the SDK function
    assert_raise('APIException'){ 
      self.class.controller.create_bulk_sms(body, content_type, accept)
    }

    # Test response code
    assert_equal(@response_catcher.response.status_code, 500)
  end

end

File: D7SMS-IOS/D7SMS/Tests/TestAPIController.m
//
//  D7SMS
//
//
#import <XCTest/XCTest.h>
#import "APIController.h"
#import "TestHelper.h"
#import "TestBaseController.h"

@interface  TestAPIController : TestBaseController
//Controller instance for all tests
@property APIController* controller;
@end 


@implementation  TestAPIController : TestBaseController

//Setup test class
//This method is called before the invocation of each test method in the test class
- (void)setUp {
    [super setUp];
    self.controller = [[APIController alloc]init];
}

//Teardown test class
//This method is called after the invocation of each test method in the test class
- (void)tearDown {
    [super tearDown];
}	


// Check Balance    
-(void) testBalance {
    //Set up test expectations
    XCTestExpectation *expectation =[self expectationWithDescription:@"High Expectations"];

    // Set callback and perform API call
    [self.controller getBalanceWithCompletionBlock:  ^(BOOL success, HttpContext* context, NSError* error) { 
       // Test response code
        XCTAssertEqual(500, context.response.statusCode);
        

        [expectation fulfill];
    }];
    [self wait];
}

// Send SMS  to a recipient using D7 SMS Gateway    
-(void) testSendSMS {
    //Set up test expectations
    XCTestExpectation *expectation =[self expectationWithDescription:@"High Expectations"];
    // Parameters for the API call
    CreateSendSMSInput *input = [[CreateSendSMSInput alloc]init];
    input.body = (SendSMSRequest*) [APIHelper jsonDeserialize: @"{  \"to\": 971562316353,  \"from\": \"SignSMS\",  \"content\": \"Send single SMS Testing\"}"
                toClass: SendSMSRequest.class];
    input.contentType = @"application/json";
    input.accept = @"application/json";

    // Set callback and perform API call
    [self.controller createSendSMSAsyncWithCreateSendSMSInput: input completionBlock:^(BOOL success, HttpContext* context, NSError* error) { 
       // Test response code
        XCTAssertEqual(500, context.response.statusCode);
        

        [expectation fulfill];
    }];
    [self wait];
}

// Send SMS  to multiple recipients using D7 SMS Gateway    
-(void) testBulkSMS {
    //Set up test expectations
    XCTestExpectation *expectation =[self expectationWithDescription:@"High Expectations"];
    // Parameters for the API call
    BulkSMSRequest* body = (BulkSMSRequest*) [APIHelper jsonDeserialize: @"{  \"messages\": [    {      \"to\": [        \"971562316353\",        \"971562316354\",        \"971562316355\"      ],      \"content\": \"Same content goes to three numbers\",      \"from\": \"SignSMS\"    }  ]}"
                toClass: BulkSMSRequest.class];
    NSString* contentType = @"application/json";
    NSString* accept = @"application/json";

    // Set callback and perform API call
    [self.controller createBulkSMSAsyncWithBody: body contentType : contentType accept : accept  completionBlock:^(BOOL success, HttpContext* context, NSError* error) { 
       // Test response code
        XCTAssertEqual(500, context.response.statusCode);
        

        [expectation fulfill];
    }];
    [self wait];
}



@end
